core/cloudflow-akka/src/main/scala/cloudflow/akkastream/AkkaStreamletContextImpl.scala

```diff
@@ -362,18 +365,30 @@ final class AkkaStreamletContextImpl(
     // we created this file when the pod started running (see AkkaStreamlet#run)
     Files.deleteIfExists(Paths.get(s"/tmp/$streamletRef.txt"))
 
-    killSwitch.shutdown()
     import system.dispatcher
+    val controls = KafkaControls.get
+
+    // Stopping will make the consumers to not read more records
+    // https://doc.akka.io/docs/alpakka-kafka/current/consumer.html#controlled-shutdown
+    val stopConsumers = controls.map(_.stop().recover {
+      case cause =>
+        system.log.error(cause, "stopping the consumer source failed.")
+        Done
+    })
     Future
-      .sequence(
-        stoppers.get.map { f ⇒
-          f().recover {
-            case cause ⇒
-              system.log.error(cause, "onStop callback failed.")
-              Dun
+      .sequence(stopConsumers)
+      .map(_ => killSwitch.shutdown())
+      .flatMap { _ =>
+        Future.sequence(
+          stoppers.get.map { f ⇒
+            f().recover {
+              case cause ⇒
+                system.log.error(cause, "onStop callback failed.")
+                Dun
+            }
           }
-        }
-      )
+        )
+      }
```

Comment:
We can do even better here and trigger `control.shutdown()` after a short while.
Alpakka Kafka uses a stop-timeout of 30s which is a bit too long for most users.